[DOCS] Add draft of Docker instructions (#6741)
ethereum/solidity: docs/installing-solidity.rst

Docker
======

Docker images of Solidity builds are available. Use the ``stable`` tag for the
latest released version, and ``nightly`` for potentially unstable changes in the develop branch.

.. code-block:: bash

>     docker run ethereum/solc:stable

Reviewer: Perhaps already show how to use options here? I think we should explain what this example code actually does - currently we only explain how to specify the version, so it might be better to combine it with what follows below. What do you think?

Reviewer: What about `docker run ethereum/solc:stable --help`?

> You can also specify specific release build versions in the tag, for example, for the 0.5.4 release.

Reviewer 2: ```suggestion
You can also specify a release build version in the tag, for example, for the 0.5.4 release.
```

.. code-block:: bash

    docker run ethereum/solc:0.5.4

> The Docker images contains the compiler executable, so you can pass all compiler arguments to it. For example.

Reviewer 2: ```suggestion
The Docker image contains the compiler executable, so you can pass all compiler arguments to it. For example.
```

.. code-block:: bash

    docker run ethereum/solc:stable --version

> To use the Docker image to compile Solidity files on the host machine mount a

Reviewer: Perhaps mention that using the compiler in standard-json-io mode works just out of the box, so you can plug the command `docker run ethereum/solc:stable --standard-json` anywhere you would use `solc`.

local folder for input and output, and specify the contract to compile. For example.

.. code-block:: bash

>     docker run -v /local/path:/mounted/path ethereum/solc:stable -o /local/path/output --abi --bin /local/path/contract.sol

Reviewer: Shouldn't it be
```suggestion
    docker run -v /local/path:/mounted/path ethereum/solc:stable -o /local/path/output --abi --bin /mounted/path/contract.sol
```
instead? Also maybe just use `/sources` in the docker image, since that actually does not have to be specific to the user.

Author: Ahh, you're right, it's local, but the mounted path.

Docker
======

> Docker images of Solidity builds are available using the `solc` image from the `ethereum` organisation. Use the ``stable`` tag for the

Reviewer: ```suggestion
Docker images of Solidity builds are available using the ``solc`` image from the ``ethereum`` organisation. Use the ``stable`` tag for the
```

latest released version, and ``nightly`` for potentially unstable changes in the develop branch.

> The command below pulls the stable version of the `solc` image (if you don't have it already), and runs it in a new container,

Reviewer: ```suggestion
The command below pulls the stable version of the ``solc`` image (if you don't have it already), and runs it in a new container,
```

Reviewer: Please change.

The Docker image runs the compiler executable, so you can pass all compiler arguments to it.
For example, the command below pulls the stable version of the ``solc`` image (if you do not have it already),
and runs it in a new container, passing the ``--help`` argument.

.. code-block:: bash

    docker run ethereum/solc:stable --help

You can also specify release build versions in the tag, for example, for the 0.5.4 release.

.. code-block:: bash

    docker run ethereum/solc:0.5.4 --help

To use the Docker image to compile Solidity files on the host machine mount a
local folder for input and output, and specify the contract to compile. For example.

.. code-block:: bash

>     docker run -v /local/path:/sources ethereum/solc:stable -o /local/path/output --abi --bin /sources/contract.sol

Reviewer: Did you try this? Shouldn't it be `-o /sources/output`?

> You can also compile non-Solidity sources in the same way, for example to use the ``--standard-json`` argument to compile JSON.

Reviewer: ```suggestion
You can also use the standard-json interface (which is recommended if the compiler is used via tooling). In this case, it is not necessary to mount any directories.
```

.. code-block:: bash

>     docker run -v /local/path:/sources ethereum/solc:stable -o /local/path/output --abi --bin --standard-json /sources/contract.json

Reviewer: ```suggestion
    docker run ethereum/solc:stable --standard-json < input.json > output.json
```

Docker
======

> Docker images of Solidity builds are available using the ``solc`` image from the ``ethereum`` organisation. Use the ``stable`` tag for the

Reviewer: "using the ``solc`` image" sounds a little like the builds are created with the help of the ``solc`` image.

latest released version, and ``nightly`` for potentially unstable changes in the develop branch.
The command below pulls the stable version of the `solc` image (if you don't have it already), and runs it in a new container,
currently passing no input files or compiler options:

.. code-block:: bash

    docker run ethereum/solc:stable

> You can also specify specific release build versions in the tag, for example, for the 0.5.4 release.

Reviewer: specify specific?

.. code-block:: bash

    docker run ethereum/solc:0.5.4

> The Docker images contains the compiler executable, so you can pass all compiler arguments to it as an argument. For example.

Reviewer: The fact that it contains the executable does not necessarily mean that that's what is run.
